src/pages/RFOX/components/AddressSelection.tsx

```diff
@@ -177,18 +191,39 @@ export const AddressSelection: FC<AddressSelectionProps> = ({
   }, [accountIdsByAccountNumberAndChainId, stakingAssetAccountNumber])
 
   const maybeRuneAccountId = useMemo(() => {
-    if (selectedAddress) return toAccountId({ account: selectedAddress, chainId: thorchainChainId })
+    if (selectedAddress && !shouldDisableAccountDropdown)
+      return toAccountId({ account: selectedAddress, chainId: thorchainChainId })
     if (currentRuneAddress)
       return toAccountId({ account: currentRuneAddress, chainId: thorchainChainId })
     if (maybeMatchingRuneAccountId) return maybeMatchingRuneAccountId
+
     return undefined
-  }, [currentRuneAddress, maybeMatchingRuneAccountId, selectedAddress])
+  }, [
+    currentRuneAddress,
+    maybeMatchingRuneAccountId,
+    selectedAddress,
+    shouldDisableAccountDropdown,
+  ])
 
   const maybeSelectedRuneAddress = useMemo(() => {
     if (!maybeRuneAccountId) return
     return fromAccountId(maybeRuneAccountId).account
   }, [maybeRuneAccountId])
 
+  const filter = useMemo(() => ({ accountId: maybeRuneAccountId }), [maybeRuneAccountId])
+  const accountNumber = useAppSelector(state => selectAccountNumberByAccountId(state, filter))
```

Comment:
Looks like the `maybeRuneAccountId` changes above may or may not be the reason why confirm click is borked for accounts never staked before?

Reply:
Nop it's not related, going to comment on the related lines